### ISSUE TITLE
sql: `TestDistSQLReceiverReportsContention/contention=false` is flakey

### ISSUE THREAD

**Issue by ajwerner** [cockroachdb/cockroach#86112]
**Describe the problem**

[Here's a build where it failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelEssentialCi/6092794?showRootCauses=false&expandBuildChangesSection=true&expandBuildProblemsSection=true&expandBuildDeploymentsSection=true&expandBuildTestsSection=true)

**To Reproduce**
Run `./dev test pkg/sql --filter 'TestDistSQLReceiverReportsContention' --stress -v --show-logs 2>&1 | tee out`. 

[TestDistSQLReceiverReportsContention.txt](https://github.com/cockroachdb/cockroach/files/9334512/TestDistSQLReceiverReportsContention.txt)

Jira issue: CRDB-18579

**Comment by ajwerner** [comment]
https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_UnitTests/6211423?showRootCauses=false&expandBuildChangesSection=true&expandBuildProblemsSection=true&expandBuildTestsSection=true

**Comment by matthewtodd** [comment]
Bisecting on a gceworker late last week showed 8303dda9774a2a98f9fe38410789c495d71a87e0, from #81995, as the commit that introduced the flake:

```
git bisect start
git bisect good dc32c9248e5d730216653d06652d8ee045f76828
git bisect bad 21365bf129f9eb618fcf23999129d82794d76f88
git bisect run ./dev test pkg/sql:sql_test --filter 'TestDistSQLReceiverReportsContention' --stress -v --show-logs --timeout=10m
```

I'll keep looking, but do you have any thoughts, @yuzefovich?

**Comment by yuzefovich** [comment]
My guess is that some internal query experienced contention. I think we probably want to disable sampling altogether with `SET CLUSTER SETTING sql.txn_stats.sample_rate = 0` so that only the our query gets a trace - then it wouldn't matter if there is contention on internal queries since the tracing would be disabled for them and there would be no way to propagate the contention events.

**Comment by matthewtodd** [comment]
Thanks, @yuzefovich, fixed in #87115.
